### DISCUSSION TITLE
curl command with SSL cert and key working fine but same tried with c++ program is not working server report error

### DISCUSSION THREAD

**Discussion by amar0501** [curl/curl#11360] [Q&A] [answered]
my curl command which is working fine

`
curl --cert "certificate.pem" --key "private.key" --form 'file=@"/path/to/file.txt"' https://localhost:50004/test
`

but use the same in c++ program server reject the request with empty response.
sample code
`

	curl_easy_setopt(curl, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE)
	curl_easy_setopt(curl, CURLOPT_URL, "https://localhost:50004/test);
	curl_easy_setopt(curl, CURLOPT_POST, 1L);
	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L);
	curl_easy_setopt(curl, CURLOPT_SSLCERT, "certificate.pem");
	curl_easy_setopt(curl, CURLOPT_SSLKEY, "private.key");
	curl_slist* headers = nullptr;
	headers = curl_slist_append(headers, "Content-Type: multipart/form-data");
	curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);

	/* Create the form */
	curl_mime *form = curl_mime_init(curl);

	/* Fill in the file upload field */
	curl_mimepart *field = curl_mime_addpart(form);
	curl_mime_name(field, "file");
	curl_mime_filedata(field, /path/to/file.txt);

	curl_easy_setopt(curl, CURLOPT_MIMEPOST, form);

`
**IF SSL part is removed it is working fine, the file is transferred to the sever, But we want to use SSL for security reason**

please help me to find the problem in my code OR do i am missing any param in my code.

**Comment by jzakrzewski** [comment]
Those two do not do the same.
You really shouldn't use -X because it is confusing.
Your command line sends a GET request with changed method string, while your code sends a real post request.

**Reply by amar0501** [reply]
updated the command and code as per your suggestion.

**Comment by bagder** [comment, marked as answer]
1. when converting a curl command line to code, the recommended starting point is `--libcurl template.c` 
2. you want the `curl_mine_*` API, not the deprecated `curl_formadd`
3. please [don't use -X POST](https://daniel.haxx.se/blog/2015/09/11/unnecessary-use-of-curl-x/) when doing HTTP POST

**Reply by amar0501** [reply]
changed the code as per suggestions, still the same problem exist.
